src/api/goods.ts

```
@@ -3,14 +3,47 @@ import { Good } from '../types/Good';
 // eslint-disable-next-line
 const API_URL = `https://mate-academy.github.io/react_dynamic-list-of-goods/goods.json`;
 
-export function getAll(): Promise<Good[]> {
-  return fetch(API_URL).then(response => response.json());
+// Função para simular atraso, útil para exibir Loader
+const wait = (ms: number) => new Promise(resolve => setTimeout(resolve, ms));
+
+// Retorna a lista completa de produtos
+export async function getAllGoods(): Promise<Good[]> {
+  try {
+    const res = await fetch(API_URL);
+    const data = await res.json();
+
+    await wait(500); // delay artificial para mostrar Loader
+
+    return Array.isArray(data) ? data : [];
+  } catch {
+    return [];
```

High-severity: getAllGoods swallows all errors and returns an empty array in the catch block. This prevents the UI from detecting real network/HTTP failures. The checklist explicitly requires: "3. [API] - When you make a request to the server, don't forget to handle possible errors" . Instead of returning [], check the HTTP status (res.ok) after fetch and throw an Error for non-OK responses; and rethrow (or throw) in the catch so the caller (App.tsx) can handle and display the error. Current code: fetch + res.json and then catch returning [] (lines below).